linera-core/src/local_node.rs

```diff
@@ -595,7 +595,11 @@ where
     where
         A: LocalValidatorNode + Clone + 'static,
     {
-        match node.download_blob(blob_id).await.map(Blob::into_hashed) {
+        match node
+            .download_blob(blob_id)
+            .await
+            .map(Blob::into_user_hashed)
```

Comment:
Here it won't always be a _user_ blob, though. Let's make this future-proof and combine the blob with the requested ID. (And still compare the hash.)

Reply:
Right, good point, let me change it